src/python_testing/TC_XX_1_1.py

```diff
@@ -0,0 +1,138 @@
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+import logging
+
+from matter_testing_support import MatterBaseTest, async_test_body, default_matter_test_main
+from mobly import asserts
+
+standard_range_prefix = 0x0000
+manufacturer_code_min_prefix = 0x0001
+manufacturer_code_max_prefix = 0xFFF0
+test_vendor_min_prefix = 0xFFF1
+test_vendor_max_prefix = 0xFFF4
+invalid_vendor_min_prefix = 0xFFF5
+invalid_vendor_max_prefix = 0xFFFF
+
+attribute_id_range_min = 0x0000
+attribute_id_range_max = 0x4FFF
+attribute_global_range_min = 0xF000
+attribute_global_range_max = 0xFFFE
+attribute_id_invalid_max = 0xFFFF
+
+# event_id_range_min = 0x00
+# event_id_range_max = 0xFF
+# event_id_invalid_max = 0xFFFF
+
+command_id_range_min = 0x00
+command_id_range_max = 0xFF
+command_id_invalid_max = 0xFFFF
+
+ATTRIBUTE_LIST_ID = 0xFFFB
+# EVENT_LIST_ID = 0xFFFA
+ACCEPTED_COMMAND_LIST_ID = 0xFFF9
+GENERATED_COMMAND_LIST_ID = 0xFFF8
+
+
+def get_element_prefix_suffix(element):
+    prefix_mask = ~0xffff
+    suffix_mask = 0xffff
+
+    prefix = (element & prefix_mask) >> 16
+    suffix = (element & suffix_mask)
+
+    return prefix, suffix
+
+
+def is_attribute_id_in_valid_range(attribute_id):
+    prefix, suffix = get_element_prefix_suffix(attribute_id)
+
+    if prefix == standard_range_prefix:
+        if suffix > attribute_id_range_max and suffix < attribute_global_range_min or suffix == attribute_id_invalid_max:
+            asserts.fail(f"Invalid attribute id {attribute_id:010x} in standard range")
+
+    elif prefix >= manufacturer_code_min_prefix and prefix <= manufacturer_code_max_prefix:
+        if suffix > attribute_id_range_max:
+            asserts.fail(f"Invalid attribute id {attribute_id:010x} in MC range")
+
+    elif prefix >= test_vendor_min_prefix and prefix <= test_vendor_max_prefix:
+        if suffix >= attribute_id_range_min:
+            asserts.fail(f"Invalid attribute id {attribute_id:010x} in Test Vendor MC range")
+
+    else:  # prefix >= invalid_vendor_min_prefix and prefix >= invalid_vendor_max_prefix
+        if suffix >= attribute_id_range_min:
+            asserts.fail(f"Invalid attribute id {attribute_id:010x} in invalid range")
+
+
+def is_command_id_in_valid_range(command_id):
+    prefix, suffix = get_element_prefix_suffix(command_id)
+
+    if prefix == standard_range_prefix:
+        if suffix > command_id_range_max:
+            asserts.fail(f"Invalid command id {command_id:010x} in standard range")
+
+    elif prefix >= manufacturer_code_min_prefix and prefix <= manufacturer_code_max_prefix:
+        if suffix > command_id_range_max:
+            asserts.fail(f"Invalid command id {command_id:010x} in MC range")
+
+    elif prefix >= test_vendor_min_prefix and prefix <= test_vendor_max_prefix:
+        if suffix >= command_id_range_min:
```

Comment:
Again, this is going to fail for all command ids, but it's obfuscating that somewhat...

Reply:
This is addressed in the latest commit